Lib/gzip.py

```diff
@@ -206,6 +221,7 @@ def __init__(self, filename=None, mode=None,
                                              zlib.DEF_MEM_LEVEL,
                                              0)
             self._write_mtime = mtime
+            self._buffer = io.BufferedWriter(_WriteBufferStream(self))
```

Comment:
Adding a `buffer_size=4*io.DEFAULT_BUFFER_SIZE` argument is probably worthwhile. The default is reasonable but not large.